### PR TITLE
Experiment

### PR DESCRIPTION
**Pull request by annelo-msft**

### CODE REVIEW

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+namespace System.ServiceModel.Rest
```
Comment (KrzysztofCwalina) [inline sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:1]: Just like in Azure.Core.dll, we should refactor types that users will be referring to from the types that are building blocks. The former should be in System.ServiceModel.Rest; the later in System.ServiceModel.Rest.Core

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    public partial class NullableResult<T>
+    {
+        public NullableResult(T? value, System.ServiceModel.Rest.Result result) { }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public virtual bool HasValue { get { throw null; } }
+        public virtual T? Value { get { throw null; } }
+        public virtual System.ServiceModel.Rest.Result GetRawResult() { throw null; }
+    }
+    public abstract partial class PipelineOptions
```
Comment (KrzysztofCwalina) [inline sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:21]: do we need this to be abstract?

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    public abstract partial class RequestBody : System.IDisposable
+    {
+        protected RequestBody() { }
+        public abstract void Dispose();
+        public abstract bool TryComputeLength(out long length);
+        public abstract void WriteTo(System.IO.Stream stream, System.Threading.CancellationToken cancellation);
+        public abstract System.Threading.Tasks.Task WriteToAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellation);
+    }
+    public partial class RequestOptions : System.ServiceModel.Rest.PipelineOptions
```
Comment (KrzysztofCwalina) [inline sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:33]: Can we just have PipelineOptions (and add CancellationToken to it)

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    public partial class Result<T> : System.ServiceModel.Rest.NullableResult<T>
+    {
+        public Result(T value, System.ServiceModel.Rest.Result result) : base (default(T), default(System.ServiceModel.Rest.Result)) { }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool HasValue { get { throw null; } }
+        public override T Value { get { throw null; } }
+        public override System.ServiceModel.Rest.Result GetRawResult() { throw null; }
+    }
+    public partial class ServiceErrorException : System.Exception
```
Comment (KrzysztofCwalina) [inline sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:55]: We should think of a better name. This sounds like the server is broken. Maybe RequestErrorException? Not great either, but probably better than ServiceErrorException

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:1]: Cool, will fix.  I'll assume these should parallel the `Azure` and `Azure.Core` namespace groupings, but let me know if not.

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 2 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:21]: Interesting!  Let me look.

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:55]: How do you feel about ResponseErrorException?  It's basically that what's coming back in the response is an error, unless we throw one of these on retry timeouts as well, e.g. when there is no response that comes back.  I'll do a quick rename to RequestErrorException for now, but we can iterate as needed.

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 3 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs]: Yep, updated

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs]: Updated

File: sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net461.cs:1]: Updated

**Comment by annelo-msft** [comment]
Closing in favor of https://github.com/Azure/azure-sdk-for-net/pull/38804

**Pull request closed without merging**